workflow/setup_expt.py

```diff
@@ -73,41 +64,36 @@ def _update_defaults(dict_in: dict) -> dict:
     # _update_defaults replaces any keys/values in defaults with matching keys in base
     yaml_dict = _update_defaults(yaml_dict)
 
-    # Override the YAML defaults with the host-specific capabilities
-    # First update config.base
-    edit_baseconfig(host, inputs, yaml_dict)
-
-    # Update stage config
-    stage_dict = {
-        "@ICSDIR@": inputs.icsdir
-    }
-    host_dict = get_template_dict(host.info)
-    stage_dict = dict(stage_dict, **host_dict)
-    stage_input = f'{inputs.configdir}/config.stage_ic'
-    stage_output = f'{inputs.expdir}/{inputs.pslot}/config.stage_ic'
-    edit_config(stage_input, stage_output, host_dict, stage_dict)
-
-    # Loop over other configs and update them with defaults
-    for cfg in yaml_dict.keys():
-        if cfg == 'base':
-            continue
-        cfg_file = f'{inputs.expdir}/{inputs.pslot}/config.{cfg}'
-        cfg_dict = get_template_dict(yaml_dict[cfg])
-        edit_config(cfg_file, cfg_file, host_dict, cfg_dict)
+    # Need to map inputs_dict keys to keys used in configs
+    inputs_dict_remapped = map_inputs_to_configs(inputs)
+
+    # Combine host.info and inputs_dict into a single dict
+    combined_dict = AttrDict(host.info, **inputs_dict_remapped)
+    combined_dict.HOMEgfs = _top
+    combined_dict.MACHINE = host.machine.upper()
+
+    files = [ff for ff in os.listdir(inputs.configdir) if os.path.isfile(os.path.join(inputs.configdir, ff))]
+    for file in files:
+        if file.endswith('.j2'):
+            logger.info(f'Jinja2 template found: {file}')
+            input_template = f'{inputs.configdir}/{file}'
+            cfg_file = file[:-3]  # remove the .j2 extension
+            output_config = f'{inputs.expdir}/{inputs.pslot}/{cfg_file}'  # output file in EXPDIR
+            cfg_key = '.'.join(cfg_file.split('.')[1:])  # key to look for in yaml_dict
+            _data = combined_dict
+            if cfg_key in yaml_dict.keys():
+                _data = AttrDict(_data, **yaml_dict[cfg_key])
```

Comment:
Alright, I see the fix now.  The host info needs to come in after the yaml dictionary:

```suggestion
    combined_dict = AttrDict(inputs_dict_remapped)
    combined_dict.HOMEgfs = _top
    combined_dict.MACHINE = host.machine.upper()

    files = [ff for ff in os.listdir(inputs.configdir) if os.path.isfile(os.path.join(inputs.configdir, ff))]
    for file in files:
        if file.endswith('.j2'):
            logger.info(f'Jinja2 template found: {file}')
            input_template = f'{inputs.configdir}/{file}'
            cfg_file = file[:-3]  # remove the .j2 extension
            output_config = f'{inputs.expdir}/{inputs.pslot}/{cfg_file}'  # output file in EXPDIR
            cfg_key = '.'.join(cfg_file.split('.')[1:])  # key to look for in yaml_dict
            _data = combined_dict
            if cfg_key in yaml_dict.keys():
                _data = AttrDict(_data, **yaml_dict[cfg_key])
                _data = AttrDict(_data, **host.info)  # Overwrite YAML defaults with host capabilities
```

This suggestion works on Gaea-C6 and all XMLs are identical to develop's.

Reply:
I have cleaned up the logic in this part of the code.  It was confusing and was doing repeated stuff.
In a later issue, we can separate the `Features` available on the `host` and ensure they are checked separately against the `base` provided `DO_` variables.

Reply:
@DavidHuber-NOAA Please take a look and let me know what you think.